syntax = "proto3";


message QromaHeartbeatUpdate {
  string heartbeatMessage = 1;
  uint32 uptime = 2;
}


message HelloQromaRequest {
  string name = 1;
}

message HelloQromaResponse {
  string response = 1;
  uint32 callCount = 2;
  uint32 nameLength = 3;
}


enum MathOperation {
  MathOp_NotSet = 0;
  MathOp_Add = 1;
  MathOp_Subtract = 2;
  MathOp_Add_And_Subtract = 3;
}

message MathRequest {
  uint32 a = 1;
  uint32 b = 2;
  MathOperation op = 3;
}

message MathResult_Add {
  uint32 result = 1;
}

message MathResult_Subtract {
  uint32 result = 1;
}

message MathResult_AddAndSubtract {
  uint32 addResult = 1;
  uint32 subtractResult = 2;
}

message MathResponse {
  oneof response {
    MathResult_Add addResult = 1;
    MathResult_Subtract subtractResult = 2;
    MathResult_AddAndSubtract addAndSubtractResult = 3;
  }
}
